packages/nx/src/command-line/sync/sync.ts

```diff
@@ -52,29 +55,108 @@ export function syncHandler(options: SyncOptions): Promise<number> {
       return 0;
     }
 
+    const {
+      failedGeneratorsCount,
+      areAllResultsFailures,
+      anySyncGeneratorsFailed,
+    } = processSyncGeneratorResultErrors(results);
+    const failedSyncGeneratorsFixMessageLines =
+      getFailedSyncGeneratorsFixMessageLines(results, options.verbose);
+
+    if (areAllResultsFailures) {
+      // if all sync generators failed to run we can't say for sure if the workspace is out of sync
+      // because they could have failed due to a bug, so we print a warning and exit with code 0
```

Comment:
It is exiting with code 1

Reply:
yeah, the comment is outdated, I'll amend it